include/extractor/internal_extractor_edge.hpp

```diff
@@ -142,14 +142,16 @@ struct InternalExtractorEdge
     }
 };
 
-struct CmpEdgeByInternalStartThenInternalTargetID
+struct CmpEdgeByInternalStartThenInternalTargetThenNameID
 {
     using value_type = InternalExtractorEdge;
     bool operator()(const InternalExtractorEdge &lhs, const InternalExtractorEdge &rhs) const
     {
         return (lhs.result.source < rhs.result.source) ||
                ((lhs.result.source == rhs.result.source) &&
-                (lhs.result.target < rhs.result.target));
+                (lhs.result.target < rhs.result.target ||
+                 (lhs.result.target == rhs.result.target &&
+                  lhs.result.name_id > rhs.result.name_id)));
```

Comment:
the parsing happens asynchronously, doesn't it? So the edges arrive in a different order which should also result in name_ids being assigned randomly? Or am I missing something? NameID should not be possible to use here?